From 97e7ab65051c90a561476bcf69a1d35aae01896c Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 1 Feb 2018 09:33:36 +0000
Subject: [PATCH] wr-policy: allow kmod_t search tracefs

Fix error while enabling wifi device on intel-x86-64:
  Failed to create system directory mac80211

Remove avc denial:
  avc: denied { search } for pid=234 comm="modprobe" \
  name="events" dev="tracefs" ino=1133 \
  scontext=system_u:system_r:kmod_t:s15:c0.c1023 \
  tcontext=system_u:object_r:tracefs_t:s0 tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/modutils.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/modutils.te b/policy/modules/system/modutils.te
index 34b133a..ae4c803 100644
--- a/policy/modules/system/modutils.te
+++ b/policy/modules/system/modutils.te
@@ -95,6 +95,7 @@ files_manage_kernel_modules(kmod_t)
 
 fs_getattr_xattr_fs(kmod_t)
 fs_dontaudit_use_tmpfs_chr_dev(kmod_t)
+fs_search_all(kmod_t)
 
 init_rw_initctl(kmod_t)
 init_use_fds(kmod_t)
-- 
2.13.3

